frameworks/flaml/exec.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import math
+import os
+
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from amlb.datautils import reorder_dataset
+from amlb.results import NoResultError, save_predictions
+from amlb.utils import dir_of, path_from_split, run_cmd, split_path, Timer
+
+from frameworks.shared.callee import save_metadata
+
+import logging
+import os
+os.environ['OMP_NUM_THREADS'] = '1'
+os.environ['OPENBLAS_NUM_THREADS'] = '1'
+os.environ['MKL_NUM_THREADS'] = '1'
+os.environ['NUMEXPR_NUM_THREADS'] = '1'
+import tempfile as tmp
+import warnings
+from flaml import AutoML
+from frameworks.shared.callee import call_run, result, output_subdir, utils
+from amlb.utils import Timer
+import pandas as pd
+import time, threading, _thread
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset: Dataset, config: TaskConfig):
+    log.info("\n**** FLAML ****\n")
```

Comment:
can you please call `save_metadata(config)` immediately after this line: I want to have a clear view of the `config` object state when it is saved, and therefore it should be done as early as possible.

Also, if you have a way to obtain the runtime version of the `flaml` package and include this in the log message + metadata (see other frameworks for examples), this would be great.

Reply:
revised. Thank you!